docs/content/dagster-cloud/developing-testing/branch-deployments/using-branch-deployments-with-github.mdx

```diff
@@ -0,0 +1,302 @@
+---
+title: Using Branch Deployments with GitHub Actions | Dagster Cloud
+description: Develop and test in the cloud.
+---
+
+# Using Branch Deployments with GitHub Actions
+
+<Note>This guide is applicable to Dagster Cloud.</Note>
+
+In this guide, we'll walk you through setting up [Branch Deployments](/dagster-cloud/developing-testing/branch-deployments) with GitHub Actions.
+
+**If you don't use GitHub for version control**, check out the dedicated [Branch deployments with TODO guide](/dagster-cloud/developing-testing/branch-deployments/using-branch-deployments).
+
+---
+
+## Prerequisites
+
+Utilizing Branch Deployments requires setting up two components: the Branch Deployment agent and continuous integration (CI) with GitHub Actions. You will need:
+
+- **Editor** or **Admin** permissions in Dagster Cloud
+- The ability to run a new agent in your infrastructure
+- The ability to configure GitHub Actions for your repository
+
+---
+
+## Step 1: Generate a Dagster Cloud agent token
+
+In this step, you'll generate a token for the agent. The Dagster Cloud agent will use this to authenticate to the agent API.
+
+1. Sign in to your Dagster Cloud instance.
+2. Click the **user menu (your icon) > Cloud settings**.
+3. In the **Cloud settings** page, click the **Configure tokens** tab.
+4. Click the **+ Create agent token** button.
+5. After the token has been created, click **Reveal token**.
+
+Keep the token somewhere handy - you'll need it to complete the setup.
+
+---
+
+## Step 2: Create and configure an agent
+
+While you can use your existing production agent, we recommend creating a dedicated branch deployment agent. This ensures that your production instance isn't negatively impacted by the workload associated with branch deployments.
+
+<details>
```

Comment:
Putting content like these in more visible tabs, once the component for it is built.